app/models/product.rb

```diff
@@ -0,0 +1,19 @@
+# == Schema Information
+#
+# Table name: products
+#
+#  id          :integer          not null, primary key
+#  name        :string
+#  price       :decimal(, )
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  category_id :integer
+#  description :text
+#  url         :string
+#
+
+class Product < ActiveRecord::Base
+  belongs_to :category
+
+  scope :with_category_id, -> (category_id) { where(category_id: category_id) }
```

Comment:
what if we put category_id param to nil? Should we then return all the products, or only the products that have a nil as a category?  should we allow product category to be nil? Nothing major just something to think about